### DISCUSSION TITLE
Notifications & CLI errors & network port confusion in documentation

### DISCUSSION THREAD

**Discussion by psyray** [reconmap/reconmap#93] [Q&A] [answered]
Hi,

I followed [the doc](https://docs.reconmap.com/admin-manual/debian-nginx-letsencrypt.html) to install an instance on a debian server.
Take apart the bad copy/paste reported [here](https://github.com/reconmap/reconmap/discussions/70#discussioncomment-2024420), no problem, reconmap is running fine, except two things, notifications and cli

In the console log I find this error for the notifications
![image](https://user-images.githubusercontent.com/1230954/150656107-4c70953f-2e2b-4244-9ad2-6551a4b48262.png)
And this one for the CLI
![image](https://user-images.githubusercontent.com/1230954/150656421-6ebc9aff-101e-4204-80cb-367f405542d7.png)

According to your reply in the discussion of the [CORS problem](https://github.com/reconmap/reconmap/discussions/70), I want to set up the right URL in the `environment.js`

But I'm stuck with the network port configuration, something weird is reported in the doc and I want to look into with you.

In the [network port](https://docs.reconmap.com/admin-manual/network-ports.html) page of the docs you write
![image](https://user-images.githubusercontent.com/1230954/150655965-e1082039-f7af-4d9f-b536-d7689a523165.png)

But in the docker-compose.yml [you setup the agent config](https://github.com/reconmap/reconmap/blob/c2aad17fef7357709f29e002691e490be25723c7/docker-compose.yml#L59) on 5520 port
![image](https://user-images.githubusercontent.com/1230954/150656460-104bac61-2c14-44f7-9da4-f8e541b2c58e.png)

So there is a conflict here, agent and notifications are on the same port.
It's a little confusing

My questions are :
- What is the good port ?
- Should I change, in the docker-compose.yml, the agent port to 5530 ?
- And the notification port ? No config in docker-compose.yml ?
- Should I create also a subdomain for the notification ? for the agent ?
- And, lastly, what url should I put into the `environment.js`
Mine is actually
![image](https://user-images.githubusercontent.com/1230954/150656341-735040ec-301b-4be7-96fb-7163a1ea7379.png)

**Comment by psyray** [comment]
@santiagolizardo ping ?
I really want to fully test your awesome app, but I'm stuck with this.
Thanks

**Reply by psyray** [reply]
According to the environment.js in your demo, I've set the 2 localhost in the config above, to the main domain, the one I use to access reconmap dashboard.
But same problem.

**Comment by santiagolizardo** [comment, marked as answer]
Hi @psyray ; It's not well documented, but the agent and notifications service are the same, hence you see them with the same port in that example. You can optionally deploy them separately in cases where you want notifications running in one box and agent in another.
Aside of that, the problem you are getting is due to the secure websocket. Because the `SECURE_TRANSPORT_ENABLED` setting is set to true, it tries to use `wss:// ` but you are probably pointing to the service directly. If you were to put it behind an nginx server and terminate the SSL there it would work. I can try to document it to make things simpler.
Hope that helps a bit. Thanks

**Reply by psyray** [reply]
Thanks for your help 😉

**Reply by psyray** [reply]
And lastly, if I understand well, in the debian install docs it miss the web socket configuration in nginx, right ?

**Reply by psyray** [reply]
Web socket seems to work with this nginx reverse proxy config on the main host (dashboard host)
```
location /notifications/ {
    proxy_pass http://127.0.0.1:5520;
    proxy_http_version 1.1;
    proxy_set_header Upgrade $http_upgrade;
    proxy_set_header Connection "Upgrade";
    proxy_set_header Host $host;
}
location /term/ {
    proxy_pass http://127.0.0.1:5520;
    proxy_http_version 1.1;
    proxy_set_header Upgrade $http_upgrade;
    proxy_set_header Connection "Upgrade";
    proxy_set_header Host $host;
}
```

**Comment by santiagolizardo** [comment]
I am closing this as the platform has changed significantly since this was last discussed. Now the agents that allow browser terminals to run are not fixed like before and you can have as many as them as needed, and they are configured through the UI. Thanks
